### DISCUSSION TITLE
Support for custom columns - like tenancy

### DISCUSSION THREAD

**Discussion by renepardon** [spatie/laravel-settings#91] [Ideas]
I'm interested in using this package on production but for that it's required I can save the settings for every tenant. An additional tenant_id column (no matter if string/UUID or uint/ID) would be nice to reference the proper settings for every tenant.

**Comment by renepardon** [comment]
I saw already https://github.com/spatie/laravel-settings/discussions/72 and https://github.com/spatie/laravel-settings/discussions/39
But in our scenario we have a single database for different reasons instead of one database per tenant. Therefore we need an extra column (tenant_id).

So, should I just create a custom model/repository and set the property dynamically on model creation or can we add support for more custom columns. And if so, how would you implement that?

**Comment by renepardon** [comment]
Meanwhile I created a custom Setting entity, migration and DatabaseSettingsRepository class to inject the tenant_id.

**Reply by frknasir** [reply]
What are the steps please?

**Reply by filip-paral** [reply]
@renepardon I know its a long time but could you point me on how you did it?

**Reply by renepardon** [reply]
Hey @frknasir @filip-paral 

I can't even remember which project it was :D 
But the migration should add the additional field(s) like tenant_id and the custom repository class extends the existing one but overwrites those methods, where tenant_id should be respected as well.

**Comment by rubenvanassche** [comment]
I think that's the correct solution, that's how I would solve it at least

**Comment by tristanjahier** [comment]
Anyway, I think the library should have native support for metadata and custom columns. Custom columns would be useful for storing the id of the user who updated the settings for example.

As of now, we need an extra database query to get these metadata (e.g.: `updated_at`), which is a pity. ☹️
